napari/layers/image/image.py

```diff
@@ -483,24 +499,27 @@ def _set_view_slice(self):
                 * self._transforms['tile2data'].scale
             )
 
-            image = np.asarray(
-                self._data_pyramid[level][tuple(indices)]
-            ).transpose(order)
+            image = np.asarray(self.data[level][tuple(indices)]).transpose(
+                order
+            )
 
             # Slice thumbnail
             indices = np.array(self.dims.indices)
             downsampled_indices = (
-                indices[not_disp] / self.downsample_factors[-1, not_disp]
+                indices[not_disp]
+                / self.downsample_factors[self._thumbnail_level, not_disp]
             )
             downsampled_indices = np.round(
                 downsampled_indices.astype(float)
             ).astype(int)
             downsampled_indices = np.clip(
-                downsampled_indices, 0, self.level_shapes[-1, not_disp] - 1
+                downsampled_indices,
+                0,
+                self.level_shapes[self._thumbnail_level, not_disp] - 1,
```

Comment:
I thought all this stuff was going to get simplified/unified in this PR? ;)

Reply:
Ah yeah, I'm saving that for another PR - turned out not to be necessary (I thought more would change in this one) so worth making that clean up a self contained thing